src/txn/txn_log.c

```diff
@@ -548,10 +548,13 @@ __wt_txn_checkpoint_log(WT_SESSION_IMPL *session, bool full, uint32_t flags, WT_
         txn->ckpt_nsnapshot = txn->snapshot_count;
         recsize = (size_t)txn->ckpt_nsnapshot * WT_INTPACK64_MAXSIZE;
         WT_ERR(__wt_scr_alloc(session, recsize, &txn->ckpt_snapshot));
-        p = txn->ckpt_snapshot->mem;
-        end = p + recsize;
-        for (i = 0; i < txn->snapshot_count; i++)
-            WT_ERR(__wt_vpack_uint(&p, WT_PTRDIFF(end, p), txn->snapshot[i]));
+        end = p = txn->ckpt_snapshot->mem;
+        /* There many not be any snapshot entries. */
+        if (end != NULL) {
+            end += recsize;
+            for (i = 0; i < txn->snapshot_count; i++)
```

Comment:
Interesting that this comes up now. This code has not changed in 8 years! (Sep 2013). If there are no snapshot entries, then we'd never use `end` because we'll skip the loop. This change seems fine anyway.